### ISSUE TITLE
Implement Unique Identifiers for Workouts and Plans

### ISSUE THREAD

**Issue by alanlowzies** [AY2122S2-CS2113T-T09-2/tp#91]
Currently, workouts are referenced by their relative indexes as seen in `workout /list`. This can be potentially an issue as the plans and schedule features are implemented.

**Example of Issue**
Assume our current workout list is as follows:
```
1. push up (10 reps)
2. lunge (20 reps)
3. russian twist (1000 reps)
4. squat (30 reps)
```

Now, suppose the user creates a plan that references indexes 1, 2, and 4 as seen from `workout /list`. Thus, the plan will look something like this:
```
1. push up (10 reps)
2. lunge (20 reps)
4. squat (30 reps)
```
> Note: This does not reflect the final visualisation of the 'plans' feature.

Next, suppose the user decides to delete Workout 3 (Russian twist with 1000 reps). So, the list will now look something like this:
```
1. push up (10 reps)
2. lunge (20 reps)
3. squat (30 reps)
```

This becomes an issue because the plan that was created references indexes 1, 2, and 4, but index 4 no longer exists, resulting in inconsistencies.

Thus, this is a motivation for workouts and plans to have unique identifiers so that plans can reference to unique identifiers of workouts. Likewise, a schedule can reference to unique identifiers of plans, instead of relative indexes that may change.

**Comment by alanlowzies** [comment]
Initial idea is to use the `HashMap` class in Java. However, `LinkedHashMap` may be a better alternative due to the predictable iteration order.

**Comment by alanlowzies** [comment]
Unique IDs for Workout objects have been implemented in PR #95.

**Comment by alanlowzies** [comment]
> Initial idea is to use the `HashMap` class in Java. However, `LinkedHashMap` may be a better alternative due to the predictable iteration order.

Currently, the `HashMap` class is being used to maintain simplicity.

**Comment by Musfirahe0556596** [comment]
Unique IDs for Plan objects have been implemented in PR #97
